### PR TITLE
Addressed further review comments

### PR DESCRIPTION
**Pull request by tomncooper**
Addresses comments from Sam H and Jakub

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java [hunk 1]
```diff
@@ -195,16 +203,57 @@ public void onClose(KubernetesClientException e) {
...
+     */
+    private Condition getCCStatusCondition(KafkaRebalanceStatus status) {
+        if (status.getConditions() != null) {
+            for (Condition condition : status.getConditions()) {
+                if (condition.getType() != null) {
+                    String conditionType = condition.getType();
+                    for (State state : State.values()) {
+                        if (conditionType.equals(state.toString())) {
+                            return condition;
+                        }
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java]: do you think we can have something using the "stream" API?

File: install/cluster-operator/050-Deployment-strimzi-cluster-operator.yaml [hunk 2]
```diff
@@ -19,7 +19,8 @@ spec:
       serviceAccountName: strimzi-cluster-operator
       containers:
       - name: strimzi-cluster-operator
-        image: strimzi/operator:latest
+        image: tomncooper/operator:cc
```
Comment (ppatierno) [inline install/cluster-operator/050-Deployment-strimzi-cluster-operator.yaml]: of course these changes should not be committed!!! :D

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java (hunk 1 above)
Comment (tomncooper) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java]: So something like:
```
if (condition.getType() != null) {
    String conditionType = condition.getType();
    if (Stream.of(State.values()).anyMatch(state -> conditionType.equals(state.toString()))) {
        return condition;
    }
}
```

**Review by kyguy — APPROVED** [review]

File: api/src/main/java/io/strimzi/api/kafka/model/status/KafkaRebalanceStatus.java [hunk 3]
```diff
@@ -29,6 +29,8 @@
 @ToString(callSuper = true)
 public class KafkaRebalanceStatus extends Status {
 
+    public static final String REBALANCE_STATUS_CONDITION_TYPE = "Status";
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/status/KafkaRebalanceStatus.java:32]: we are calling it "State" not "Status" as the enum for representing ... states of the state machine.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java [hunk 4]
```diff
@@ -195,16 +205,62 @@ public void onClose(KubernetesClientException e) {
...
+     * @throws RuntimeException If there is more than one Condition instance in the supplied status with the type
+     *                          {@link KafkaRebalanceStatus#REBALANCE_STATUS_CONDITION_TYPE}.
+     */
+    private Condition getRebalanceStateCondition(KafkaRebalanceStatus status) {
+        if (status.getConditions() != null) {
+
+            List<Condition> statusConditions = status.getConditions()
+                    .stream()
+                    .filter(condition -> condition.getType() != null)
+                    .filter(condition -> condition.getType().equals(KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_TYPE))
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:225]: just one `filter` with following check:

```java
condition -> condition.getType() != null &&
          condition.getType().equals(KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_TYPE)
```

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java [hunk 5]
```diff
@@ -426,7 +482,14 @@ public void onClose(KubernetesClientException e) {
...
             // stay in the current error state, actually failing the Future
-            return Future.failedFuture(kafkaRebalance.getStatus().getConditions().get(0).getMessage());
+            Condition statusCondition = getRebalanceStateCondition(kafkaRebalance.getStatus());
+            String statusMessage;
+            if (statusCondition != null) {
+                statusMessage = statusCondition.getMessage();
+            } else {
+                statusMessage = "Error retrieving status message";
+            }
+            return Future.failedFuture(statusMessage);
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:492]: one line

```java
return Future.failedFuture(statusCondition != null ? statusCondition.getMessage() : "Error retrieving status message");
```

File: operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java [hunk 6]
```diff
@@ -28,23 +28,27 @@ public static String iso8601Now() {
...
-    public static Condition buildConditionFromException(Throwable error) {
-        return buildCondition(error == null ? "Ready" : "NotReady", error);
+    public static Condition buildConditionFromException(String type, String conditionStatus, Throwable error) {
+        return buildCondition(type, conditionStatus, error);
     }
 
-    public static Condition buildCondition(String type, Throwable error) {
+//    public static Condition buildCondition(String type, Throwable error) {
+//        return buildCondition(type, "True", error);
+//    }
```
Comment (ppatierno) [inline operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java:37]: remove this

**Review by ppatierno — COMMENTED** [review]
LGTM! I did a little bit of minor refactoring changes and opened a new PR #68 to replace this one.

**Pull request closed without merging**